webapp/src/js/services/contact-view-model-generator.js

```diff
@@ -211,6 +213,29 @@ angular.module('inboxServices').factory('ContactViewModelGenerator',
       });
     };
 
+    var getHeading = function(report) {
+      if (report.validSubject) {
+        return report.subject.value;
+      }
+      if (report.subject.name) {
+        return report.subject.name;
+      }
+      return $translate.instant('report.subject.unknown');
```

Comment:
This code will error if the report isn't found. Is that possible, or will GetDataRecords error out first? Maybe it's worth adding a check that says 

```js
if (!report) {
  return $translate.instant('report.subject.unknown');
}
```